### PR TITLE
Feature/catx a1 sldt@a1 sldt 285 aas

### PR DESCRIPTION
**Pull request by fmisir**
This PR replaces the Digital Twin Registry API with the AAS Registry API.

All AAS API methods are implemented.

Following is minor topics are not implemented:

 - The AAS API  does not provide any field length (min/max) information. We need to define our own.
 - Provide a custom, more client - convenient fetch API

I will organise a session where we can decide on that. The PR can be reviewed regardless as the changes will be minimal.

### CODE REVIEW

File: semantics/services/src/main/resources/db/changelog/db.changelog-v1.yaml [hunk 1]
```diff
@@ -0,0 +1,270 @@
+databaseChangeLog:
```
Comment (LuLeRoemer) [inline semantics/services/src/main/resources/db/changelog/db.changelog-v1.yaml:1]: Do you also delete the old DB tables or should we do that in a next step?

**Review by LuLeRoemer — COMMENTED** [review]
I had some problems with the lookup endpoint. In what format do you expect the assetIds. I tried multiple ways but only got 500 or 400.
Also, do we need pagination in the lookup endpoint?

File: semantics/services/src/main/resources/db/changelog/db.changelog-v1.yaml (hunk 1 above)
Comment (fmisir) [reply semantics/services/src/main/resources/db/changelog/db.changelog-v1.yaml:1]: I would suggest to drop the database - so that we start on a fresh one.

**Comment by fmisir** [comment]
> I had some problems with the lookup endpoint. In what format do you expect the assetIds. I tried multiple ways but only got 500 or 400. Also, do we need pagination in the lookup endpoint?

You have to provide an asset id object. I'am currently updating the API Spec with a better example.
Meanwhile you can have a look at the test case here. The request will be written to the logs so that you can take it from there.
https://github.com/catenax/tractusx/pull/409/files#diff-066e0ad07717d920ade18578925dcb8bda3b394e1ea43570c7b89384fed32bbcR521

File: semantics/services/pom.xml [hunk 2]
```diff
@@ -48,7 +48,7 @@
     <dependencies>
         <dependency>
             <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-data-jpa</artifactId>
+            <artifactId>spring-boot-starter-data-jdbc</artifactId>
```
Comment (drcgjung) [inline semantics/services/pom.xml:51]: ok, so Semantic Hub Persistence was already migrated to graphdb.

File: semantics/services/src/main/java/net/catenax/semantics/registry/controller/AssetAdministrationShellApiDelegate.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    @Override
+    public ResponseEntity<SubmodelDescriptor> getSubmodelDescriptorById(String aasIdentifier, String submodelIdentifier) {
+        Submodel submodel = shellService.findSubmodelByExternalId(aasIdentifier, submodelIdentifier);
+        return new ResponseEntity<>(submodelMapper.toApiDto(submodel), HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<AssetAdministrationShellDescriptor> postAssetAdministrationShellDescriptor(AssetAdministrationShellDescriptor assetAdministrationShellDescriptor) {
```
Comment (drcgjung) [inline semantics/services/src/main/java/net/catenax/semantics/registry/controller/AssetAdministrationShellApiDelegate.java:95]: we would need a ResponseEntity<List<AASD>> postAssetAdministrationShellDescriptors(List<AssetAdministrationShellDescriptor> descs,boolean override) batch create endpoint 
- override true|false (true see put, false throws error on override attempt)

File: semantics/services/src/main/java/net/catenax/semantics/registry/controller/AssetAdministrationShellApiDelegate.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        Shell saved = shellService.save(shellMapper.fromApiDto(assetAdministrationShellDescriptor));
+        return new ResponseEntity<>(shellMapper.toApiDto(saved), HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<SubmodelDescriptor> postSubmodelDescriptor(String aasIdentifier, SubmodelDescriptor submodelDescriptor) {
```
Comment (drcgjung) [inline semantics/services/src/main/java/net/catenax/semantics/registry/controller/AssetAdministrationShellApiDelegate.java:101]: we would need a ResponseEntity<Map<String,List<SubmodelDescriptor>>> postSubmodelDescriptors(Map<String,List<SubmodelDescriptor>>,boolean override) batch endpoint

File: semantics/services/src/main/java/net/catenax/semantics/registry/controller/AssetAdministrationShellApiDelegate.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    @Override
+    public ResponseEntity<Void> deleteAllAssetLinksById(String aasIdentifier) {
+        shellService.deleteAllIdentifiers(aasIdentifier);
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    @Override
+    public ResponseEntity<List<String>> getAllAssetAdministrationShellIdsByAssetLink(List<IdentifierKeyValuePair> assetIds) {
```
Comment (drcgjung) [inline semantics/services/src/main/java/net/catenax/semantics/registry/controller/AssetAdministrationShellApiDelegate.java:129]: We would need a ResponseEntity<List<SubmodelDescriptor>> getAllSubmodelDescriptorsByAssetLink(List<IdentifiedKeyValuePair> assetIds, List<String> submodelIdentifiers)

File: semantics/services/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+      name: uuid_function
+      value: uid.uuid_generate_v4()
+      dbms: postgres
+  - property:
+      name: uuid_function
+      value: random_uuid()
+      dbms: h2
+  - include:
+      file: db.changelog-v1.yaml
+      relativeToChangelogFile: true
```
Comment (drcgjung) [inline semantics/services/src/main/resources/db/changelog/db.changelog-master.yaml:16]: there should be a newline

**Review by drcgjung — APPROVED** [review]
Batch Create needed
Batch Fetch needed

GlobalAssert-Thingy I will overtake that

**Comment by fmisir** [comment]
@drcgjung 

I introduced the validation logic for duplicates and field lengths. All test cases do also run against postgresql (tested with 13.5).
From my point of view we can ship this version.

As we discussed, i will create the tasks for the additional features you mentioned:

Batch Create needed
Batch Fetch needed
GlobalAssert-Thingy

**Review by drcgjung — APPROVED** [review]
LGTM

**Pull request merged**